Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -506,13 +506,16 @@ function Get-MachinesFqdnsForLB
         #Map the public ip id to the fqdn
         foreach($publicIp in $publicIPAddressResources)
         {
-            if(-not [string]::IsNullOrEmpty($publicIP.DnsSettings.Fqdn))
+            if(-not [string]::IsNullOrEmpty($publicIp.IpConfiguration.Id))
```

Comment:
If all the publicIP in the loop turns out to be bad, you will exit with no entry in fqdnMap. Azure File copy needs at-least one valid IP or DNS to progress. Where is that check?